version: '3'
services:
  auth_database:
    image: postgres:17-alpine
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=1
      - POSTGRES_DB=authdb
    ports:
      - "5433:5432"
    volumes:
    - ./auth_service/init_db.sql:/docker-entrypoint-initdb.d/init_db.sql
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U postgres"]
      interval: 5s
      timeout: 5s
      retries: 10
  auth_service:
    build: 
      context: ./auth_service
      dockerfile: Dockerfile
    ports:
      - "8000:8000"
    depends_on:
      auth_database:
        condition: service_healthy
    environment:
      - DATABASE_URL=postgresql://postgres:1@auth_database:5432/authdb
    